test/testfloat/CMakeLists.txt

```diff
@@ -22,63 +22,72 @@ if(TESTFLOAT_GEN)
+    # than required by the WebAssembly and the IEEE-754 specs.
+    set(IGNORE_NANS FALSE)
+
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         # TODO: Clang produces -0 for input 0, see https://bugs.llvm.org/show_bug.cgi?id=47393.
         list(APPEND IGNORE_LIST ui64_to_f64/min)
+    elseif(CMAKE_CROSSCOMPILING AND CMAKE_SYSTEM_PROCESSOR STREQUAL arm64)
+        # Ignore NaNs in arithmetic functions.
+        set(IGNORE_NANS TRUE)
     endif()
 
     set(ROUNDING_MODES near_even minMag min max)
 
-    set(
-        ROUNDING_DEPENDENT_FUNCTIONS
+    # Arithmetic functions: sensitive to rounding and NaNs.
+    set(ARITHMETIC_FUNCTIONS
         f32_sqrt f64_sqrt
         f32_add f64_add
         f32_sub f64_sub
         f32_mul f64_mul
-        f32_div f64_div
+        f32_div f64_div)
 
-        i32_to_f32 ui32_to_f32 i64_to_f32 ui64_to_f32 i64_to_f64 ui64_to_f64
-        f64_to_f32
-    )
+    # Comparison functions: insensitive to rounding and NaNs.
+    set(COMPARISON_FUNCTIONS
+        f32_eq f64_eq f32_lt f64_lt f32_le f64_le)
 
-    set(
-        ROUNDING_INDEPENDENT_FUNCTIONS
-        f32_eq f64_eq f32_lt f64_lt f32_le f64_le
+    # Conversion functions sensitive to rounding.
+    set(INEXACT_CONVERSION_FUNCTIONS
+        i32_to_f32 ui32_to_f32 i64_to_f32 ui64_to_f32 i64_to_f64 ui64_to_f64
+        f64_to_f32)
 
-        # Always exact -> no rounding
+    # Conversion functions which are always exact -> no rounding.
+    set(EXACT_CONVERSION_FUNCTIONS
         i32_to_f64 ui32_to_f64
-        f32_to_f64
-    )
+        f32_to_f64)
 
-    set(
-        TRUNC_FUNCTIONS
-        f32_trunc_to_i32 f32_trunc_to_ui32 f32_trunc_to_i64 f32_trunc_to_ui64 f64_trunc_to_i32 f64_trunc_to_ui32 f64_trunc_to_i64 f64_trunc_to_ui64
-    )
+    # Truncate conversion functions - the minMag rounding mode is forced.
+    set(TRUNC_FUNCTIONS
+        f32_trunc_to_i32 f32_trunc_to_ui32 f32_trunc_to_i64 f32_trunc_to_ui64
+        f64_trunc_to_i32 f64_trunc_to_ui32 f64_trunc_to_i64 f64_trunc_to_ui64)
 
     # Here we need to map Wasm function to testfloat_gen arguments as Wasm has multiple functions
     # with non-sticky rounding direction while testfloat_gen has single function and rounding
     # direction options.
     #
-    # TODO: NaNs are ignored for these as Fizzy always return nan:canonical.
-    set(
-        ROUND_TO_INT_FUNCTIONS
+    # TODO: NaNs are always ignored for these as Fizzy always return nan:canonical.
+    set(ROUND_TO_INT_FUNCTIONS
         "f32_ceil\;-rmax f32_roundToInt"
         "f32_floor\;-rmin f32_roundToInt"
         "f32_trunc\;-rminMag f32_roundToInt"
         "f32_nearest\;-rnear_even f32_roundToInt"
         "f64_ceil\;-rmax f64_roundToInt"
         "f64_floor\;-rmin f64_roundToInt"
         "f64_trunc\;-rminMag f64_roundToInt"
-        "f64_nearest\;-rnear_even f64_roundToInt"
-    )
+        "f64_nearest\;-rnear_even f64_roundToInt")
 
+    string(REPLACE ";" " " EMULATOR "${CMAKE_CROSSCOMPILING_EMULATOR}")
```

Comment:
Can you explain why is this replacing needed?

Reply:
Explained.

Reply:
But why can't it be with spaces from the beginning?

Reply:
It must be "a semicolon-separated list, then the first value is the command and remaining values are its arguments."
https://cmake.org/cmake/help/latest/prop_tgt/CROSSCOMPILING_EMULATOR.html